packages/authenticator/amplify_authenticator/lib/src/widgets/social/social_button.dart

```diff
@@ -81,24 +86,37 @@ class SocialSignInButton extends AuthenticatorButton<SocialSignInButton> {
   const SocialSignInButton({
     super.key,
     required this.provider,
+    this.icon,
     this.maxWidth = double.infinity,
   });
 
   /// A social sign-in button for Facebook.
   const SocialSignInButton.facebook({Key? key})
-      : this(key: key, provider: AuthProvider.facebook);
+      : this._(key: key, provider: AuthProvider.facebook);
 
   /// A social sign-in button for Google.
   const SocialSignInButton.google({Key? key})
-      : this(key: key, provider: AuthProvider.google);
+      : this._(key: key, provider: AuthProvider.google);
 
   /// A social sign-in button for Amazon.
   const SocialSignInButton.amazon({Key? key})
-      : this(key: key, provider: AuthProvider.amazon);
+      : this._(key: key, provider: AuthProvider.amazon);
 
   /// A social sign-in button for Apple.
   const SocialSignInButton.apple({Key? key})
-      : this(key: key, provider: AuthProvider.apple);
+      : this._(key: key, provider: AuthProvider.apple);
+
+  const SocialSignInButton._({
+    super.key,
+    required this.provider,
+    this.maxWidth = double.infinity,
+    this.icon,
+  });
+
+  /// Icon for the social provider. If omitted, default social providers (Amazon,
+  /// Apple, Facebook or Google) will render appropriate icons automatically. Custom
+  /// providers with no defined icon will not render an icon.
+  final Widget? icon;
```

Comment:
The rest is an implementation detail since you can only set the icon for custom providers, and these docs will show when hovering over the property in the constructor.

```suggestion
  /// Icon for the social provider.
  final Widget? icon;
```